smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareExecutionGenerator.kt

```diff
@@ -44,8 +47,13 @@ class MiddlewareExecutionGenerator(
         val httpMethod = resolveHttpMethod(op)
 
         // FIXME it over indents if i add another indent, come up with better way to properly indent or format for swift
-        writer.write("  .withEncoder(value: encoder)")
-        writer.write("  .withDecoder(value: decoder)")
+
+        if (ctx.service.requestWireProtocol != WireProtocol.XML) {
+            writer.write("  .withEncoder(value: encoder)")
+        }
+        if (ctx.service.responseWireProtocol != WireProtocol.XML) {
+            writer.write("  .withDecoder(value: decoder)")
+        }
```

Comment:
Encoders are only added to context for an operation when the service needs them.